desktop/core/src/desktop/js/reactComponents/MetricsComponent/MetricsComponent.tsx

```diff
@@ -0,0 +1,149 @@
+import './MetricsComponent.scss';
+
+const { Option } = Select;
+
+const MetricsComponent: React.FC = (): JSX.Element => {
+  const [metrics, setMetrics] = useState<MetricsResponse>();
+  const [loading, setLoading] = useState(true);
+  const [error, setError] = useState<string>();
+  const [searchQuery, setSearchQuery] = useState<string>('');
+  const [selectedMetric, setSelectedMetric] = useState<string>('');
+  const [showAllTables, setShowAllTables] = useState(true);
+  const [filteredMetricsData, setFilteredMetricsData] = useState<MetricsData[]>([]);
+  const dropdownRef = useRef(null);
+
+  useEffect(() => {
+    const fetchData = async () => {
+      try {
+        const response = await get<MetricsResponse>('/desktop/metrics/', { format: 'json' });
+        setMetrics(response);
+      } catch (error) {
+        setError(error.message);
+      } finally {
+        setLoading(false);
+      }
+    };
+
+    fetchData();
+  }, []);
+
+  useEffect(() => {
+    if (!metrics) {
+      return;
+    }
+
+    const filteredData = parseMetricsData(metrics).filter(tableData =>
+      tableData.dataSource.some(data => data.name.toLowerCase().includes(searchQuery.toLowerCase()))
+    );
+
+    setFilteredMetricsData(filteredData);
+  }, [searchQuery, metrics]);
+
+  const parseMetricsData = (data: MetricsResponse) => {
+    return Object.keys(data.metric)
+      .filter(
+        key =>
+          !key.startsWith('auth') &&
+          !key.startsWith('multiprocessing') &&
+          !key.startsWith('python.gc')
+      )
+      .map(key => ({
+        caption: key,
+        dataSource: Object.keys(data.metric[key]).map(subKey => ({
+          name: subKey,
+          value: data.metric[key][subKey]
+        }))
+      }));
+  };
+  const handleMetricChange = (value: string) => {
+    setSelectedMetric(value);
+    setShowAllTables(value === '');
+  };
+
+  const handleFilterInputChange = (e: React.ChangeEvent<HTMLInputElement>) => {
+    setSearchQuery(e.target.value);
+  };
+
+  return (
+    <div className="cuix antd metrics-component">
+      <Spin spinning={loading}>
+        {!error && (
+          <>
+            <Input
+              className="metrics-filter"
+              placeholder="Filter metrics..."
+              value={searchQuery}
+              onChange={handleFilterInputChange}
+              prefix={<SearchOutlined />}
+            />
+
+            <Select
+              className="metrics-select"
+              //to make sure antd class gets applied
+              getPopupContainer={triggerNode => triggerNode.parentElement}
+              ref={dropdownRef}
+              value={selectedMetric}
+              onChange={handleMetricChange}
+              data-testid="metric-select"
+            >
+              <Option value="">All</Option>
+              {metrics &&
+                Object.keys(metrics.metric)
+                  .filter(
+                    key =>
+                      !key.startsWith('auth') &&
```

Comment:
as we have a filtering the data anyways outside in useEffect anyways, better to these filtering there itself mad just render the filterer data.